src/panels/lovelace/cards/hui-fan-card.ts

```diff
@@ -0,0 +1,357 @@
+                })}
+                @action=${this._handleAction}
+                .actionHandler=${actionHandler({
+                  hasHold: hasAction(this._config!.hold_action),
+                  hasDoubleClick: hasAction(this._config!.double_tap_action),
+                })}
+                tabindex="0"
+              ></ha-icon-button>
+            </div>
+          </div>
+
+          <div id="info">
+            ${UNAVAILABLE_STATES.includes(stateObj.state)
+              ? html`
+                  <div>
+                    ${computeStateDisplay(
+                      this.hass.localize,
+                      stateObj,
+                      this.hass.language
+                    )}
+                  </div>
+                `
+              : html` <div class="speed"></div> `}
+            ${this._config.name || computeStateName(stateObj)}
+          </div>
+        </div>
+      </ha-card>
+    `;
+  }
+
+  protected shouldUpdate(changedProps: PropertyValues): boolean {
+    return hasConfigOrEntityChanged(this, changedProps);
+  }
+
+  protected updated(changedProps: PropertyValues): void {
+    super.updated(changedProps);
+    if (!this._config || !this.hass) {
+      return;
+    }
+
+    const stateObj = this.hass!.states[this._config!.entity];
+
+    if (!stateObj) {
+      return;
+    }
+
+    const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
+    const oldConfig = changedProps.get("_config") as FanCardConfig | undefined;
+
+    if (
+      !oldHass ||
+      !oldConfig ||
+      oldHass.themes !== this.hass.themes ||
+      oldConfig.theme !== this._config.theme
+    ) {
+      applyThemesOnElement(this, this.hass.themes, this._config.theme);
+    }
+  }
+
+  private _dragEvent(e: any): void {
+    const stateObj = this.hass!.states[this._config!.entity] as FanEntity;
+    if (e.detail.value < 1) {
+      this.shadowRoot!.querySelector(".speed")!.innerHTML = `Off`;
+    } else {
+      this.shadowRoot!.querySelector(".speed")!.innerHTML = `${
+        stateObj.attributes.speed_list[e.detail.value - 1]
+      }`;
+    }
+    this._showspeed();
+    this._hidespeed();
+  }
+
+  private _showspeed(): void {
+    clearTimeout(this._speedTimeout);
+    this.shadowRoot!.querySelector(".speed")!.classList.add("show_speed");
+  }
+
+  private _hidespeed(): void {
+    this._speedTimeout = window.setTimeout(() => {
+      this.shadowRoot!.querySelector(".speed")!.classList.remove("show_speed");
+    }, 500);
+  }
+
+  private _setSpeed(e: any): void {
+    const stateObj = this.hass!.states[this._config!.entity] as FanEntity;
+    if (e.detail.value < 1) {
+      this.hass!.callService("fan", "turn_off", {
+        entity_id: this._config!.entity,
+      });
+    } else {
+      this.hass!.callService("fan", "turn_on", {
+        entity_id: this._config!.entity,
+        speed: stateObj.attributes.speed_list[e.detail.value - 1],
+      });
+    }
+  }
+
+  private _computeSpeed(stateObj: FanEntity): string {
```

Comment:
This can probably be done inline in the HTML template with a ternary

```ts
stateObj.state === "off" || !stateObj.attributes.speed_list ? "off" : `${stateObj.attributes.speed}`
```